onnxruntime/core/providers/webnn/builders/impl/ternary_op_builder.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Intel Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <core/providers/common.h>
+
+#include "core/providers/webnn/builders/model_builder.h"
+#include "core/providers/webnn/builders/op_builder_factory.h"
+#include "core/providers/webnn/builders/helper.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace webnn {
+
+class TernaryOpBuilder : public BaseOpBuilder {
+  // Add operator related.
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+};
+
+// Add operator related.
+
+Status TernaryOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                                              const logging::Logger& /* logger */) const {
+  const auto& op_type(node.OpType());
+
```

Comment:
```suggestion
  ORT_RETURN_IF(node.InputDefs().size() < 3, "Operator requires at least three inputs");
  
```

I'd just add a safety check here.